website/src/controllers/groups.js

```diff
@@ -30,13 +30,15 @@ var guildPopulate = {path: 'members', select: nameFields, options: {limit: 15} }
  * limited fields - and only a sampling of the members, beacuse they can be in the thousands
  * @param type: 'party' or otherwise
  * @param q: the Mongoose query we're building up
- * @param additionalFields: if we want to populate some additional field not fetched normally 
+ * @param additionalFields: if we want to populate some additional field not fetched normally
  *        pass it as a string, parties only
  */
-var populateQuery = function(type, q, additionalFields){
+var populateQuery = function(type, q, additionalFields, user){
   if (type == 'party')
     q.populate('members', partyFields + (additionalFields ? (' ' + additionalFields) : ''));
   else
+    //Use Conditional Semantics to always include the user
+    guildPopulate.match =  {"_id": { "$ne" : null, "$in": [user._id] } };
```

Comment:
Can you confirm that public guilds you visit when you are not a member of them still show a members list?